frontend/document/nondocument.lua

```diff
@@ -0,0 +1,165 @@
+--[[--
+NonDocument
+
+any file *with extension* can become a nondocument
+if there's one or more providers that know how to open it.
+
+@module nondocument
+]]
+
+local Blitbuffer = require("ffi/blitbuffer")
+local ButtonDialogTitle = require("ui/widget/buttondialogtitle")
+local Document = require("document/document")
+local DocumentRegistry = require("document/documentregistry")
+local RenderImage = require("ui/renderimage")
+local Screen = require("device").screen
+local UIManager = require("ui/uimanager")
+local logger = require("logger")
+local util = require("util")
+local _ = require("gettext")
+
+local function getIcon(path, width, height, zoom)
+    local bb, is_straight_alpha = RenderImage:renderSVGImageFile(path, width, height, zoom)
+    local icon = Blitbuffer.new(width, height, Blitbuffer.TYPE_BBRGB32)
+    icon:fill(Blitbuffer.COLOR_WHITE)
+    if is_straight_alpha then
+        if Screen.sw_dithering then
+            icon:ditheralphablitFrom(bb, 0, 0, 0, 0, icon.w, icon.h)
+        else
+            icon:alphablitFrom(bb, 0, 0, 0, 0, icon.w, icon.h)
+        end
+    else
+        if Screen.sw_dithering then
+            icon:ditherpmulalphablitFrom(bb, 0, 0, 0, 0, icon.w, icon.h)
+        else
+            icon:pmulalphablitFrom(bb, 0, 0, 0, 0, icon.w, icon.h)
+        end
+    end
+    return icon
+end
+
+local NonDocument = Document:extend{
+    _document = false,
+    provider = "nondocument",
+    provider_name = _("File Manager"),
+
+    extensions = {},
+    providers = {},
+}
+
+function NonDocument:_readMetadata() return true end
+
+function NonDocument:init() end
+function NonDocument:close() end
+function NonDocument:register() end
+function NonDocument:getPageCount() end
+
+function NonDocument:getProps()
+    local _, _, docname = self.file:find(".*/(.*)")
+    docname = docname or self.file
+    return {
+        title = docname:match("(.*)%."),
+    }
+end
+
+function NonDocument:getCoverPageImage()
+    local extension = util.getFileNameSuffix(self.file)
+    for __, v in ipairs(self.extensions[extension]) do
+        if v.icon_path then
+            return getIcon(v.icon_path, 100, 100, 4)
```

Comment:
@poire-z ·@NiLuJe: any magic trick to put sensible values here for width, height and zoom factor?

I've got current values by trial and error and, even if they work on **my** emulator I'm not sure that's good enough.

Reply:
No obvious idea.
Even if it's not really an icon, and it may get big in Mosaic view, just do as in:
https://github.com/koreader/koreader/blob/741562e84c2a33e38858ef379b3db95406a75a79/frontend/ui/widget/iconwidget.lua#L39-L40
and zoom nil.